### DISCUSSION TITLE
Need help in creating stack SLC and geometry information using the miaplpy and the ISCE

### DISCUSSION THREAD

**Discussion by kangahdesmond** [luhipi/sarvey#78] [Q&A]
I have been able to install the SARvey and run the sample data and everything works perfectly but am left with how to to do the preparation to get the two files to be used by SARvey, PLease i need help.

**Comment by Andreas-Piter** [comment]
Hi Desmond,
Thanks for your interest in SARvey. I am happy to hear that you succesfully run the sample dataset.
To create your own dataset, please follow the [instructions](https://github.com/isce-framework/isce2/blob/main/contrib/stack/README.md) for creating an SLC stack with [ISCE](https://github.com/isce-framework/isce2). Once your coregistered SLC stack is ready, you can load the data into the format specific to [Miaplpy](https://github.com/insarlab/MiaplPy) which is described [here](https://luhipi.github.io/sarvey/docs/preparation.html#loading-data-into-miaplpy).
Best, Andreas

**Comment by kangahdesmond** [comment]
Thank you very much but the installation of this two has been my problem

Sent from my T-Mobile 5G Device
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by kangahdesmond** [comment]
Good day brother, please with the results of ISCE to used by the miaplpy, can SNAP produce that ?

________________________________

**Comment by Andreas-Piter** [comment]
Dear @kangahdesmond, SARvey relies on the preprocessors supported by MiaplPy. Currently, only ISCE and GAMMA are supported. We are still looking for someone who is familiar enough with SNAP to provide the support for SNAP inside MiaplPy. 
Best, Andreas

**Comment by kangahdesmond** [comment]
Thanks

Sent from my T-Mobile 5G Device
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by Basin-Ir** [comment]
Thank you very much for developing this software. I installed SARvey on Windows 11 Pro using WSL-Ubuntu and ran it with the Masjed Soleyman dataset provided in the software’s demo. It worked perfectly, and the outputs were generated successfully.

However, I am also facing issues with ISCE. It is indeed a very complex package with limited documentation. It seems that although it installs on the Linux distribution within WSL, it fails to run properly. It constantly throws errors, even though the installation completed successfully and the package was imported correctly in the Python environment. The errors appear to be related to incomplete compilation of certain functions during installation.

I wanted to ask whether ISCE2 has been tested in a similar setup on your side. How should it actually be used in practice? Is there any alternative solution? The most helpful support would be if you could provide the raw Sentinel data for the Masjed Soleyman case starting from the ISCE processing stage.

Thank you very much.

**Reply by mahmud1** [reply]
> Thank you very much for developing this software. I installed SARvey on Windows 11 Pro using WSL-Ubuntu and ran it with the Masjed Soleyman dataset provided in the software’s demo. It worked perfectly, and the outputs were generated successfully.
> 
> However, I am also facing issues with ISCE. It is indeed a very complex package with limited documentation. It seems that although it installs on the Linux distribution within WSL, it fails to run properly. It constantly throws errors, even though the installation completed successfully and the package was imported correctly in the Python environment. The errors appear to be related to incomplete compilation of certain functions during installation.
> 
> I wanted to ask whether ISCE2 has been tested in a similar setup on your side. How should it actually be used in practice? Is there any alternative solution? The most helpful support would be if you could provide the raw Sentinel data for the Masjed Soleyman case starting from the ISCE processing stage.
> 
> Thank you very much.

Thank you for your message and for testing SARvey. Currently, we rely on MiaplPy to import data into SARvey. Miaplpy supports ISCE2 (free) and GAMMA (commercial) and we are working on adding support for SNAP in the future. The installation and processing up to Miaplpy are independent of SARvey, so could be done in a separate setup if it is simpler. For details about [ISCE2](https://github.com/isce-framework/isce2) and [MiaplPy](https://github.com/insarlab/MiaplPy), please refer to their repositories and documentation. There is also the [ISCE+ course](https://www.earthscope.org/event/insar-processing-and-analysis-isce/) with many learning materials and videos you can use to learn ISCE. Once you have the data in the format SARvey accepts, you can start with SARvey. You can also find more sample data processed with ISCE on [MiaplPy repository](https://github.com/insarlab/MiaplPy/tree/main/docs). The same data can be used in SARvey as well.
